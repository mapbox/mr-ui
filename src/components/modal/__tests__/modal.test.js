import React from 'react';
import { shallow } from 'enzyme';
import { noDisplayCases } from './modal-test-cases';

describe('Modal', () => {
  let testCase;
  let wrapper;

  describe(noDisplayCases.basicSmall.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.basicSmall;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('fires onExit', () => {
      // find the button that triggers onExit method
      wrapper.find({ 'data-test': 'modal-close' }).prop('onClick')();

      expect(testCase.props.onExit).toHaveBeenCalledTimes(1);
    });

    test('allowEventBubbling disables event trapping', () => {
      expect(wrapper.find('EventTrap').length).toBe(1);
      wrapper.setProps({ allowEventBubbling: true });
      expect(wrapper.find('EventTrap').length).toBe(0);
    });
  });

  describe(noDisplayCases.basicLarge.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.basicLarge;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('fires onExit', () => {
      // find the button that triggers onExit method
      wrapper.find({ 'data-test': 'modal-close' }).prop('onClick')();

      expect(testCase.props.onExit).toHaveBeenCalledTimes(1);
    });
  });

  describe(noDisplayCases.basicAuto.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.basicAuto;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('fires onExit', () => {
      // find the button that triggers onExit method
      wrapper.find({ 'data-test': 'modal-close' }).prop('onClick')();

      expect(testCase.props.onExit).toHaveBeenCalledTimes(1);
    });
  });

  describe(noDisplayCases.allOptions.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.allOptions;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe(noDisplayCases.optionalOnExit.description, () => {
    beforeEach(() => {
      testCase = noDisplayCases.optionalOnExit;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(wrapper).toMatchSnapshot();
    });

    test('close button is not present', () => {
      expect(wrapper.find({ 'data-test': 'modal-close' }).exists()).toBe(false);
    });
  });
});
