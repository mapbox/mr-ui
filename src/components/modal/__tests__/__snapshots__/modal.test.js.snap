// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal No fixed width, display only renders 1`] = `
<Displaced
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  onExit={[MockFunction]}
  titleText="Auto modal title"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm "
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full  bg-white round px36 py36"
  >
    <div>
      No fixed width on the modal container. But very tall.
    </div>
    <div
      className="absolute top right"
    >
      <Tooltip
        alignment="center"
        block={true}
        coloring="light"
        content="Close"
        disabled={false}
        maxWidth="medium"
        padding="small"
        placement="top"
        respondsToClick={false}
        textSize="s"
      >
        <button
          aria-label="Close"
          className="btn btn--transparent unround-t unround-br color-gray py12 px12"
          data-test="modal-close"
          onClick={[MockFunction]}
          type="button"
        >
          <Icon
            inline={false}
            name="close"
            passthroughProps={Object {}}
            size={18}
          />
        </button>
      </Tooltip>
    </div>
  </div>
</Displaced>
`;

exports[`Modal all options renders 1`] = `
<Displaced
  getApplicationNode={[Function]}
  initialFocus="#foo"
  mounted={true}
  onExit={[MockFunction]}
  titleText="All options"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm "
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w360 bg-white round px36 py36"
  >
    <div>
      <div
        className="mb12"
      >
        I am a message
      </div>
      <button
        aria-label="press me"
        className="btn"
        id="foo"
      >
        press me
      </button>
    </div>
    <div
      className="absolute top right"
    >
      <Tooltip
        alignment="center"
        block={true}
        coloring="light"
        content="Close"
        disabled={false}
        maxWidth="medium"
        padding="small"
        placement="top"
        respondsToClick={false}
        textSize="s"
      >
        <button
          aria-label="Close"
          className="btn btn--transparent unround-t unround-br color-gray py12 px12"
          data-test="modal-close"
          onClick={[MockFunction]}
          type="button"
        >
          <Icon
            inline={false}
            name="close"
            passthroughProps={Object {}}
            size={18}
          />
        </button>
      </Tooltip>
    </div>
  </div>
</Displaced>
`;

exports[`Modal basic default renders 1`] = `
<Displaced
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  onExit={[MockFunction]}
  titleText="Large modal title"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm "
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w600 bg-white round px36 py36"
  >
    <div>
      Large modal body
    </div>
    <div
      className="mt24"
    >
      <ModalActions
        primaryAction={
          Object {
            "callback": [MockFunction],
            "destructive": true,
            "text": "Okay",
          }
        }
        secondaryAction={
          Object {
            "callback": [MockFunction],
            "text": "Cancel",
          }
        }
        tertiaryAction={
          Object {
            "callback": [MockFunction],
            "text": "Give up",
          }
        }
      />
    </div>
    <div
      className="absolute top right"
    >
      <Tooltip
        alignment="center"
        block={true}
        coloring="light"
        content="Close"
        disabled={false}
        maxWidth="medium"
        padding="small"
        placement="top"
        respondsToClick={false}
        textSize="s"
      >
        <button
          aria-label="Close"
          className="btn btn--transparent unround-t unround-br color-gray py12 px12"
          data-test="modal-close"
          onClick={[MockFunction]}
          type="button"
        >
          <Icon
            inline={false}
            name="close"
            passthroughProps={Object {}}
            size={18}
          />
        </button>
      </Tooltip>
    </div>
  </div>
</Displaced>
`;

exports[`Modal basic small renders 1`] = `
<Displaced
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  onExit={[MockFunction]}
  titleText="Small modal title"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm "
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w360 bg-white round px36 py36"
  >
    <div>
      Small modal body
    </div>
    <div
      className="mt24"
    >
      <ModalActions
        primaryAction={
          Object {
            "callback": [MockFunction],
            "text": "Okay",
          }
        }
      />
    </div>
    <div
      className="absolute top right"
    >
      <Tooltip
        alignment="center"
        block={true}
        coloring="light"
        content="Close"
        disabled={false}
        maxWidth="medium"
        padding="small"
        placement="top"
        respondsToClick={false}
        textSize="s"
      >
        <button
          aria-label="Close"
          className="btn btn--transparent unround-t unround-br color-gray py12 px12"
          data-test="modal-close"
          onClick={[MockFunction]}
          type="button"
        >
          <Icon
            inline={false}
            name="close"
            passthroughProps={Object {}}
            size={18}
          />
        </button>
      </Tooltip>
    </div>
  </div>
</Displaced>
`;

exports[`Modal no onExit renders 1`] = `
<Displaced
  focusDialog={true}
  getApplicationNode={[Function]}
  mounted={true}
  titleText="No onExit passed as prop"
  underlayClass="bg-darken50 px12 py12 px60-mm py60-mm "
  underlayProps={
    Object {
      "data-popover-ignore-clicks": true,
    }
  }
  underlayStyle={
    Object {
      "zIndex": 1,
    }
  }
>
  <div
    className="relative wmax-full w360 bg-white round px36 py36"
  >
    <div>
      You can't close me
    </div>
  </div>
</Displaced>
`;
